tests/python_functional/functional_tests/destination_drivers/snmp_destination/001-snmp-destination-v2c-normal-working-compared-cisco-ios-traps-default-values.py

```diff
@@ -0,0 +1,68 @@
+#############################################################################
+# Copyright (c) 2020 One Identity
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+import os
+
+import pytest
+
+
+CISCO_TIMETICKS = '97881'
+CISCO_FACILITY = "SYS"
+CISCO_SEVERITY = "6"
+CISCO_MESSAGE_TYPE = "CONFIG_I"
+CISCO_MESSAGE = "Configured from console by vty1 (10.30.0.32)"
+
+input_log = "<38>Feb 11 21:27:22 testhost testprogram[9999]: test message\n"
+expected_log = 'iso.3.6.1.4.1.9.9.41.1.2.3.1.2.55 = STRING: "' + CISCO_FACILITY + '"	iso.3.6.1.4.1.9.9.41.1.2.3.1.3.55 = INTEGER: ' + CISCO_SEVERITY + '	iso.3.6.1.4.1.9.9.41.1.2.3.1.4.55 = STRING: "' + CISCO_MESSAGE_TYPE + '"	iso.3.6.1.4.1.9.9.41.1.2.3.1.5.55 = STRING: "' + CISCO_MESSAGE + '"	iso.3.6.1.4.1.9.9.41.1.2.3.1.6.55 = Timeticks: (' + CISCO_TIMETICKS + ')'
+
+
+@pytest.mark.skipif(os.system("which snmptrapd") != 0, reason="snmptrapd can't found on system")
+def test_acceptance(config, syslog_ng, snmptrapd):
+    # create source driver config
+    file_source = config.create_file_source(file_name="input.log")
+
+    # create destination driver config
+    snmp_objs = (
+        "'1.3.6.1.4.1.9.9.41.1.2.3.1.2.55', 'Octetstring', '%s'" % CISCO_FACILITY,
+        "'1.3.6.1.4.1.9.9.41.1.2.3.1.3.55', 'Integer', '%s'" % CISCO_SEVERITY,
+        "'1.3.6.1.4.1.9.9.41.1.2.3.1.4.55', 'Octetstring', '%s'" % CISCO_MESSAGE_TYPE,
+        "'1.3.6.1.4.1.9.9.41.1.2.3.1.5.55', 'Octetstring', '%s'" % CISCO_MESSAGE,
+        "'1.3.6.1.4.1.9.9.41.1.2.3.1.6.55', 'Timeticks', '%s'" % CISCO_TIMETICKS,
+    )
+
+    trap_obj = "'.1.3.6.1.6.3.1.1.4.1.0','Objectid','.1.3.6.1.4.1.9.9.41.2.0.1'"
+
+    snmp_destination = config.create_snmp_destination(
+        host="'127.0.0.1'",
+        port=snmptrapd.get_port(),
+        version="'v2c'",
+        snmp_obj=snmp_objs,
+        trap_obj=trap_obj,
+    )
+
+    config.create_logpath(statements=[file_source, snmp_destination])
+    config.update_global_options(keep_hostname="yes")
+    file_source.write_log(input_log)
+
+    syslog_ng.start(config)
+
+    assert snmptrapd.wait_for_trap(expected_log)
```

Comment:
Testcase assertion should be on top level, without this the report of assertion will contain only function calls which does not help on debugging. Let move back the logic here (to the testcase)
```
    dst_file = DestinationReader(FileIO)
    assert expected_log in dst_file.read_all_logs(snmptrapd.snmptrapd_log)
```

Reply:
I don't like this idea as this is against the encapsulation concept. The snmptrapd_log and the handling of trapd's log file is an internal implementation detail of the helper class.

To help the debugging, I can put debug/error log messages into wait_for_trap function to report the details of the not matching patterns.

Reply:
Ok, I have an other suggestion. What if we change the original `wait_for_trap()` to something like this:
```
    def get_trap_logs(self):
        file_reader = DestinationReader(FileIO)
        return file_reader.read_all_logs(self.snmptrapd_log)
```

and in the testcase we can call assert like this:
```
    assert expected_log in  snmptrapd.get_trap_logs()
```

Reply:
sounds good :-)

Reply:
thanks